tests/Utils/CoerceInputValueTest.php

```diff
@@ -177,6 +178,31 @@ public function testReturnsErrorForIncorrectValueType(): void
         )]);
     }
 
+    public function testPropagatesClientSafeError(): void
+    {
+        $clientSafeException = new class('message') extends \Exception implements ClientAware {
+            public function isClientSafe(): bool
+            {
+                return true;
+            }
+        };
+
+        $scalar = new CustomScalarType([
+            'name' => 'TestScalar',
+            'parseValue' => function () use ($clientSafeException) {
+                throw $clientSafeException;
+            },
+            'parseLiteral' => fn () => null,
+        ]);
+
+        $result = Value::coerceInputValue(['value' => 1], $scalar);
+        $this->expectGraphQLError($result, [CoercionError::make(
+            'message',
+            null,
+            ['value' => 1],
```

Comment:
The expected values line up with values that are previously set up in the test. Please use variables to make the association clear.